src/main/scala/fastq.scala

```diff
@@ -269,7 +269,7 @@ case object fastq {
 
     def parseFastqPhred33: Iterator[Option[FASTQ]] =
       lines
-        .grouped(4)
+        .grouped(4).filter(_.size == 4) // grouped is broken
```

Comment:
`grouped` is not broken, it works as documented. You have an `Iterator` here, you get a `GroupedIterator`, so you can use `.withPartial(false)` instead of another pass with a filter.

And why don't you want to use https://github.com/ohnosequences/fastarious/pull/13#discussion_r102705424? It's shorter, uses meaningful names and avoids this problem.